File: src/main/java/org/example/kalkulationsprogramm/dto/EmailAbsenderDto.java
package org.example.kalkulationsprogramm.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.kalkulationsprogramm.domain.EmailAbsender;

@Data
@NoArgsConstructor
public class EmailAbsenderDto {

    private Long id;
    private String emailAdresse;
    private String anzeigename;
    private boolean aktiv = true;
    private int sortierung = 0;

    public static EmailAbsenderDto fromEntity(EmailAbsender entity) {
        if (entity == null) {
            return null;
        }
        EmailAbsenderDto dto = new EmailAbsenderDto();
        dto.setId(entity.getId());
        dto.setEmailAdresse(entity.getEmailAdresse());
        dto.setAnzeigename(entity.getAnzeigename());
        dto.setAktiv(entity.isAktiv());
        dto.setSortierung(entity.getSortierung());
        return dto;
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/EmailAbsenderDto.kt
package org.example.kalkulationsprogramm.dto

import org.example.kalkulationsprogramm.domain.EmailAbsender
import java.lang.reflect.Field

data class EmailAbsenderDto(
    var id: Long? = null,
    var emailAdresse: String? = null,
    var anzeigename: String? = null,
    var isAktiv: Boolean = true,
    var sortierung: Int = 0,
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: EmailAbsender?): EmailAbsenderDto? {
            if (entity == null) {
                return null
            }
            val dto = EmailAbsenderDto()
            dto.id = entity.readField("id") as? Long
            dto.emailAdresse = entity.readField("emailAdresse") as? String
            dto.anzeigename = entity.readField("anzeigename") as? String
            dto.isAktiv = entity.readField("aktiv") as? Boolean ?: true
            dto.sortierung = entity.readField("sortierung") as? Int ?: 0
            return dto
        }

        private fun EmailAbsender.readField(name: String): Any? {
            val field: Field = EmailAbsender::class.java.getDeclaredField(name)
            field.isAccessible = true
            return field.get(this)
        }
    }
}
